docker-compose.yml

```diff
@@ -0,0 +1,61 @@
+version: '3.7'
+
+services:
+  mysql:
+    image: mysql:8.0
+    command: --default-authentication-plugin=mysql_native_password
+    environment:
+      MYSQL_ROOT_PASSWORD: password
+    volumes:
+      - mysql:/var/lib/mysql
+
+  postgres:
+    image: postgres:13.2
+    environment:
+      POSTGRES_USER: root
+      POSTGRES_PASSWORD: password
+    volumes:
+      - postgres:/var/lib/postgresql/data
+
+  app:
+    build:
+      context: .dockerdev
+      dockerfile: Dockerfile
+      args:
+        RUBY_VERSION: "2.7.2"
+        PG_VERSION: 13
+        NODE_VERSION: 14
+        MYSQL_VERSION: "8.0"
+        BUNDLER_VERSION: 2
+    image: solidus-3.0.0.rc1
+    command: bash -c "(bundle check || bundle) && tail -f Gemfile"
+    environment:
+      CAPYBARA_DRIVER: selenium_chrome_headless_docker_friendly
+      DB_USERNAME: root
+      DB_PASSWORD: password
+      RAILS_VERSION: ${RAILS_VERSION:-~> 6.1.0}
+      DB_ALL: "1"
+      DB_MYSQL_HOST: mysql
+      DB_POSTGRES_HOST: postgres
+      HISTFILE: "/home/solidus_user/history/bash_history"
+      MYSQL_HISTFILE: "/home/solidus_user/history/mysql_history"
+    ports:
+      - "3000:3000"
+    volumes:
+      - .:/home/solidus_user/app:cached
+      - bundle:/home/solidus_user/gems
+      - history:/home/solidus_user/history
+      - .dockerdev/.psqlrc:/home/solidus_user/.psqlrc
```

Comment:
About volumes: I find it a good choice to have the project path as `delegated` and the cache paths (`gems` and `node_modules`) as `cached`. Some extra info here: https://tkacz.pro/docker-volumes-cached-vs-delegated/

Reply:
Thanks for pointing me to that article. That's interesting stuff I didn't have the necessity to study before as I'm a Linux user. But isn't it the other way round? The project path will have a lot of writing from the host, while the `GEM_PATH` will only be written by the container.

Reply:
hey @blocknotes , what do you think about it?

Reply:
From the [docs](https://docs.docker.com/engine/reference/commandline/service_create/#add-bind-mounts-volumes-or-memory-filesystems) (scroll down a bit to the consistency parameter):
_"delegated: The container runtime's view of the mount is authoritative. There may be delays before updates made in a container are visible on the host."_
So delegated is good for the project path because with it we give priority to the host changes. I understand that for a Linux user this changes little but on the Mac side, Docker volume sync slowness is an issue, unfortunately.
Cached is the opposite, so it fits well for cache paths.

Reply:
Understood, makes sense! (however, the article you pointed me takes the opposite as conclusions :see_no_evil: )

Changed!

Reply:
Ah sorry about the article. I read a nice blog post times ago about the volumes types but it has been passed a while so it's probably not that one 😓

Reply:
No worries :blush: